exporter/awsemfexporter/metric_translator.go

```diff
@@ -118,6 +134,268 @@ func (dps DoubleHistogramDataPointSlice) At(i int) DataPoint {
+		serviceNamespace, svcNsOk := rm.Resource().Attributes().Get(conventions.AttributeServiceNamespace)
+		if svcNameOk && svcNsOk && serviceName.Type() == pdata.AttributeValueSTRING && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = fmt.Sprintf("%s/%s", serviceNamespace.StringVal(), serviceName.StringVal())
+		} else if svcNameOk && serviceName.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceName.StringVal()
+		} else if svcNsOk && serviceNamespace.Type() == pdata.AttributeValueSTRING {
+			namespace = serviceNamespace.StringVal()
+		}
+	}
+
+	if len(namespace) == 0 {
+		namespace = defaultNameSpace
+	}
+	return namespace
+}
+
+// TranslateOtToGroupedMetric converts OT metrics to GroupedMetric format
+func TranslateOtToGroupedMetric(metric pdata.Metrics, config *Config) (map[string]*GroupedMetric, int) {
+	totalDroppedMetrics := 0
+	groupedMetricMap := make(map[string]*GroupedMetric)
+	var dps DataPoints
+	var instrumentationLibName string
+	rms := metric.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		if rm.IsNil() {
+			continue
+		}
+		namespace := getNamespace(&rm, config.Namespace)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ilm := ilms.At(j)
+			if ilm.IsNil() {
+				continue
+			}
+
+			if ilm.InstrumentationLibrary().IsNil() {
+				instrumentationLibName = noInstrumentationLibraryName
+			} else {
+				instrumentationLibName = ilm.InstrumentationLibrary().Name()
+			}
+
+			metrics := ilm.Metrics()
+			for k := 0; k < metrics.Len(); k++ {
+				metric := metrics.At(k)
+				if metric.IsNil() {
+					totalDroppedMetrics++
+					continue
+				}
+
+				switch metric.DataType() {
+					case pdata.MetricDataTypeIntGauge:
+						dps = IntDataPointSlice{metric.IntGauge().DataPoints()}
+					case pdata.MetricDataTypeDoubleGauge:
+						dps = DoubleDataPointSlice{metric.DoubleGauge().DataPoints()}
+					case pdata.MetricDataTypeIntSum:
+						dps = IntDataPointSlice{metric.IntSum().DataPoints()}
+					case pdata.MetricDataTypeDoubleSum:
+						dps = DoubleDataPointSlice{metric.DoubleSum().DataPoints()}
+					case pdata.MetricDataTypeDoubleHistogram:
+						dps = DoubleHistogramDataPointSlice{metric.DoubleHistogram().DataPoints()}
+					default:
+						config.logger.Warn("Unhandled metric data type.",
+							zap.String("DataType", metric.DataType().String()),
+							zap.String("Name", metric.Name()),
+							zap.String("Unit", metric.Unit()),)
+						continue
+				}
+
+				if dps.Len() == 0 {
+					continue
+				}
+
+				for m := 0; m < dps.Len(); m++ {
+					dp := dps.At(m)
+					if dp.IsNil() {
+						continue
+					}
+
+					fields := make(map[string]interface{})
+					fields[nameSpace] = namespace
+					fields[OTellibDimensionKey] = instrumentationLibName
+	
+					dp.LabelsMap().ForEach(func(k string, v string) {
+						fields[k] = v
+					})
+					
+					s := make([]string, len(fields)*2)
```

Comment:
Why did we initialize this to `len(fields)*2`? Shouldn't this be 0?
```suggestion
					s := make([]string, 0, len(fields))
```

Reply:
We can even move this up into the above `ForEach` loop instead of iterating over it twice.